csi/controller.go

```diff
@@ -944,13 +946,30 @@ func (s *OsdCsiServer) CreateSnapshot(
 			// https://github.com/kubernetes-csi/external-snapshotter/blob/v6.0.1/pkg/sidecar-controller/snapshot_controller.go#L656-L678
 			return nil, status.Errorf(codes.Aborted, "Volume id %s not found: %v", req.GetSourceVolumeId(), err)
 		}
-		return nil, status.Errorf(codes.Internal, "Failed to create snapshot: %v", err)
+
+		// Check if snapshot has been created but is in error state
+		snapInfo, errFindFailed := util.VolumeFromIdSdk(ctx, volumes, req.GetName())
+		if errFindFailed == nil && !isSnapshotReady(snapInfo) {
```

Comment:
should the `isSnapshotReady` be a loop ? what if the snap is still in process ?

Reply:
In the case that we found, this error never resolved itself after ~10 minutes:
> VOLUME	SnapshotCreateFailure		697143149459174097			ALARM		1	Feb 17 02:44:53 UTC 2023	Feb 17 02:44:53 UTC 2023	Volume (Id: 697143149459174097 Name: snapshot-04230195-be88-4064-893e-eb50c5c23c73 Source: 633915026553598074) snapshot failed with error: All nodes which store the volume data are not online or data is not yet fully replicated on all replicas. Snapshot will automatically complete when volume is ready.